packages/react-charting/src/components/VerticalBarChart/VerticalBarChartRTL.test.tsx

```diff
@@ -0,0 +1,447 @@
+        calloutProps={{ doNotLayer: true }}
+        onRenderCalloutPerDataPoint={props =>
+          props ? (
+            <div className="onRenderCalloutPerDataPoint">
+              <p>Custom Callout Content</p>
+            </div>
+          ) : null
+        }
+      />,
+    );
+    await new Promise(resolve => setTimeout(resolve));
+    const getById = queryAllByAttribute.bind(null, 'id');
+    const line = screen.getAllByText((content, element) => element!.tagName.toLowerCase() === 'path')[0];
+    expect(line).toBeDefined();
+    fireEvent.mouseOver(line);
+    await new Promise(resolve => setTimeout(resolve));
+    // Assert
+    expect(getById(container, /toolTipcallout/i)).toBeDefined();
+    expect(screen.queryByText('Custom Callout Content')).toBeNull();
+  });
+});
+
+describe('Vertical bar chart - Subcomponent xAxis Labels', () => {
+  test('Should show the x-axis labels tooltip when hovered', async () => {
+    // Arrange
+    const { container } = render(<VerticalBarChart data={pointsWithLine} showXAxisLablesTooltip={true} />);
+    await new Promise(resolve => setTimeout(resolve));
+    const getById = queryAllByAttribute.bind(null, 'id');
+    const bars = screen.getAllByText((content, element) => element!.tagName.toLowerCase() === 'rect');
+    expect(bars).toHaveLength(8);
+    fireEvent.mouseOver(bars[0]);
+    await new Promise(resolve => setTimeout(resolve));
+    // Assert
+    expect(getById(container, /showDots/i)).toHaveLength(5);
+    expect(getById(container, /showDots/i)[0]!.textContent!).toEqual('20,0...');
+  });
+
+  test('Should show rotated x-axis labels', async () => {
+    // Arrange
+    const { container } = render(<VerticalBarChart data={simplePoints} rotateXAxisLables={true} />);
+    await new Promise(resolve => setTimeout(resolve));
+    const getByClass = queryAllByAttribute.bind(null, 'class');
+    expect(getByClass(container, /tick/i)[0].getAttribute('transform')).toContain('rotate(-45)');
+  });
+});
+
+describe('Screen resolution', () => {
+  const originalInnerWidth = global.innerWidth;
+  const originalInnerHeight = global.innerHeight;
+  afterEach(() => {
+    global.innerWidth = originalInnerWidth;
+    global.innerHeight = originalInnerHeight;
+    act(() => {
+      global.dispatchEvent(new Event('resize'));
+    });
+  });
+
+  test('Should remain unchanged on zoom in', () => {
+    // Arrange
+    const { container } = render(<VerticalBarChart data={chartPoints} width={300} height={300} />);
+
+    // Act
+    global.innerWidth = window.innerWidth / 2;
+    global.innerHeight = window.innerHeight / 2;
+    act(() => {
+      global.dispatchEvent(new Event('resize'));
+    });
+
+    // Assert
+    expect(container).toMatchSnapshot();
+  });
+
+  test('Should remain unchanged on zoom out', () => {
+    // Arrange
+    const { container } = render(<VerticalBarChart data={chartPoints} width={300} height={300} />);
+
+    // Act
+    global.innerWidth = window.innerWidth * 2;
```

Comment:
some functions are getting called again and again. Like zoom in/zoom out, mouse hover. Can we put them into a utility file.